modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts

```diff
@@ -121,8 +121,8 @@ const htmlDiff: (v1: string, v2: string) => string = (() => {
       }
     } else {
       if (out.n[0].text == null) {
-        for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
-          str += '<del>' + out.o[n] + oSpace[n] + '</del>';
+        for (let j = 0; j < out.o.length && out.o[j].text == null; j++) {
```

Comment:
TypeScript 4.5 complained about this as it was able to infer the original type for `n` was a string apparently :shrug: 

```
modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124:14 - error TS2322: Type 'number' is not assignable to type 'string'.
124         for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
                 ~

modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:124:64 - error TS2356: An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.
124         for (n = 0; n < out.o.length && out.o[n].text == null; n++) {
                                                                   ~

modules/agar/src/main/ts/ephox/agar/assertions/Differ.ts:136:81 - error TS2356: An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.
136           for (n = out.n[i].row + 1; n < out.o.length && out.o[n].text == null; n++) {
                                                                                    ~
```

Reply:
Or maybe it's just more strict about not allowing changes to the type of a `let` variable